synapse/handlers/message.py

```diff
@@ -1016,9 +1048,12 @@ async def create_new_client_event(
             if already_exists:
                 raise SynapseError(400, "Can't send same reaction twice")
 
-        logger.debug("Created event %s", event.event_id)
-
-        return event, context
+        # If this relation is a thread, then ensure thread head is not part of
+        # a thread already.
```

Comment:
The MSC says that the thread head should not have any relation. (I.e. it must be an event that is not related to any other event.)